core/src/main/scala/scala/pickling/Tools.scala

```diff
@@ -220,8 +223,8 @@ trait RichTypes {
           sym.fullName +
           (if (sym.isModuleClass) ".type" else "") +
           (if (targs.isEmpty) "" else targs.map(_.key).mkString("[", ",", "]"))
-        case _ =>
-          tpe.toString
+        case t  if t.toString contains "with" => sys.error(s"Cannot create key from type: $t")
```

Comment:
This looks hacky. Not that too many class name would contain the String `with` like `withholding` in lowercase, but still.